### PR TITLE
Improvements to paginated TD retrieval spec

### PR DESCRIPTION
**Pull request by farshidtz**
- [x] Specify query arguments based on the original proposal
- [x] Specify headers and modify to match https://www.w3.org/TR/ldp-paging/ as suggested in https://github.com/w3c/wot-discovery/issues/16#issuecomment-792663791
- [x] Change spec document to match header-based approach

***
<a href="https://pr-preview.s3.amazonaws.com/farshidtz/wot-discovery/pull/130.html" title="Last updated on Mar 23, 2021, 10:17 AM UTC (d78b6ca)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-discovery/130/e6f18dd...farshidtz:d78b6ca.html" title="Last updated on Mar 23, 2021, 10:17 AM UTC (d78b6ca)">Diff</a>

### CODE REVIEW

File: directory.td.json [hunk 1]
```diff
@@ -181,6 +184,64 @@
...
+            "uriVariables": {
+                "offset": {
+                    "description": "The number of TDs to skip before the collection",
+                    "type": "number",
+                    "default": 0
+                },
+                "limit": {
+                    "description": "The number of TDs in the collection",
+                    "type": "number",
+                    "default": 1
```
Comment (farshidtz) [inline directory.td.json:198]: @sebastiankb @mmccool Is there a way to set default values in TD? 
The [URI Template](https://tools.ietf.org/html/rfc6570) does not support it after [draft v5](https://tools.ietf.org/html/draft-gregorio-uritemplate-05#section-2.5) and I couldn't find anything in the TD's DataSchema either.

**Review by AndreaCimminoArriaga — APPROVED** [review]
I'm approving the review, as only remark I would add the optional functionality to return in the header the pair "Link" and an ontology class URI (as specified in the LDPP). In this way when the TDs are returned by using the headers one could know the types of these TDs.

**Comment by farshidtz** [comment]
We could try to fully comply with [RFC7233 Range Requests](https://tools.ietf.org/html/rfc7233). This would let us query range of bytes (or range of TDs) by setting equivalent request headers.

To have dedicated resource URLs for the RESTful API, we could allow passing range request values in query. E.g.:
```
/td?range-start=0&range-end=100&range-unit=TDs gives the first 100 TDs
/td?range-start=0&range-end=100&range-unit=bytes gives the first 100 bytes
```
The above is inline with [LDP Paging](https://www.w3.org/TR/ldp-paging/). Moreover, we could borrow LDPP's Links spec to have next link and etag to identify the state.

**Comment by AndreaCimminoArriaga** [comment]
@farshidtz it makes sense to have both ranges, maybe we could even define others. I like the approach, it is sensible.

File: index.html [hunk 2]
```diff
@@ -1066,25 +1141,86 @@ <h4>Listing</h4>
...
+                                constructing the URL or by using the `next` Link from the previous response:
+                                <pre>
+                                    GET /td?offset=10&limit=10 HTTP/1.1   
+                                    Host: tdd.example.com
+                                </pre>
+
+                                Response:
+                                <pre>
+                                    HTTP/1.1 200 OK
+                                    Content-Range: TDs 10-12/*
```
Comment (danielpeintner) [inline index.html:1192]: I believe the range should be "10-11"
Otherwise it would be 3 more TDs and the count 12 does not match (0-11)

**Comment by farshidtz** [comment]
I changed this to draft. As mentioned in my last [comment](https://github.com/w3c/wot-discovery/pull/130#issuecomment-801005408), we should try to fully comply with one standard and define query arguments to make pages addressable by URL. But this makes the specification even longer and this is overkill when there are much simpler approaches; see #145.

File: directory.td.json (hunk 1 above)
Comment (mmccool) [reply directory.td.json:198]: Best to create an issue about this in the TD repo, otherwise it may not get addressed.

**Comment by mmccool** [comment]
Closing without merging, but as discussed Farshid will create a new PR with equivalent (but improved) content.

File: directory.td.json (hunk 1 above)
Comment (farshidtz) [reply directory.td.json:198]: Thanks. I'll reply on the new issue: https://github.com/w3c/wot-thing-description/issues/1093

**Pull request closed without merging**